src/lib/ring-buffer.ts

```diff
@@ -66,11 +66,11 @@ export class RingBuffer {
     let spaceToEnd = this.capacity - this.end;
     if (chunk.length <= spaceToEnd) {
       this.buffer.set(chunk, this.end);
-      this.end += chunk.length;
+      this.end = (this.end + chunk.length) % this.capacity;
```

Comment:
I don't think this change is necessary, is it? If the chunk fits within the remaining available space, we can just do addition and avoid the modulo.

Reply:
Yeah sorry, was playing around with growing the buffer differently as well.